plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java

```diff
@@ -152,6 +153,9 @@ private static Object convertValue(Type type, Object value)
         if (type instanceof RealType) {
             return intBitsToFloat(toIntExact((Long) value));
         }
+        else if (type instanceof VarcharType) {
+            return ((Slice) value).toStringUtf8();
+        }
         return value;
```

Comment:
In future we should handle all types explicitly here to avoid surprises.

Reply:
sounds good, will do!